test/testAuction.js

```diff
@@ -0,0 +1,44 @@
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+
+describe("Auction", function () {
+    let hnft;
+    let ad3;
+    let auction;
+    let owner;
+    let user;
+
+    beforeEach(async () => {
+        [owner, user] = await ethers.getSigners();
+
+        const Auction = await ethers.getContractFactory("AuctionAndMicroPayment");
+        auction = await Auction.deploy();
+        await auction.deployed();
+
+        const HNFT = await ethers.getContractFactory("ERC5489");
+        hnft = await HNFT.deploy();
+        await hnft.deployed();
+
+        const AD3 = await ethers.getContractFactory("AD3Token");
+        ad3 = await AD3.deploy(1000);
+        await ad3.deployed();
+
+        ad3.transfer(user.address, 200);
+    });
+
+    it("Should bid success", async () => {
```

Comment:
add more cases:
1. bid should fail when newBidAmount is L.T. slotBalance * 1.2
2. bid should success when newBidAmount is EQ slotBalance * 1.2

add more cases for payout.

Reply:
Repost